src/App.tsx

```diff
@@ -1,15 +1,66 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import cn from 'classnames';
 import './App.scss';
+import { Person } from './types/Person';
 import { peopleFromServer } from './data/people';
+import debounce from 'lodash.debounce';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+type Props = {
+  delay?: number;
+  onSelected?: (person: Person | null) => void;
+};
+
+export const App: React.FC<Props> = () => {
+  const [targetPerson, setTargetPerson] = useState<Person | null>();
+  const [currentTodos, setCurrentTodos] = useState(peopleFromServer);
+  const [inputSort, setInputSort] = useState<string>('');
+  const [appliedInputSort, setAppliedInputSort] = useState('');
+
+  const handleChangeInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputSort(event.target.value);
+    if (inputSort.length > 0) {
+      setCurrentTodos(
+        currentTodos.filter(person =>
+          person.name
+            .toLocaleLowerCase()
+            .includes(appliedInputSort.trim().toLocaleLowerCase()),
+        ),
+      );
+    } else {
+      setCurrentTodos(peopleFromServer);
+    }
+  };
+
+  useEffect(() => {
+    const handler = debounce(() => setAppliedInputSort(inputSort), 300);
+
+    handler();
+
+    return () => handler.cancel();
+  }, [inputSort]);
+
+  const todos = currentTodos.map((item, index) => (
```

Comment:
Don't need to use index here
```suggestion
  const todos = currentTodos.map((item) => (
```